src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/ServiceBusTriggerInput.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Microsoft.Azure.WebJobs.ServiceBus
+{
+    // The core object we get when an ServiceBus is triggered. 
+    // This gets converted to the user type (Message, string, poco, etc) 
+    internal sealed class ServiceBusTriggerInput
+    {
+        // If != -1, then only process a single message in this batch. 
+        private int _selector = -1;
+
+        public MessageReceiver MessageReceiver { get; set; }
+        public IMessageSession MessageSession { get; set; }
+
+        internal Message[] Messages { get; set; }
+
+        public bool IsSingleDispatch
```

Comment:
Based on which factory method is called (CreateSingle/CreateBatch) you'd just set this appropriately. No need for a selector.